persistence-dse-6.8/src/main/java/io/stargate/db/dse/DsePersistenceActivator.java

```diff
@@ -31,12 +35,20 @@
 
 public class DsePersistenceActivator extends BaseActivator {
 
+  private static final String AUTHZ_PROCESSOR_ID =
+      System.getProperty("stargate.authorization.processor.id");
+
   private final ServicePointer<Metrics> metrics = ServicePointer.create(Metrics.class);
   private final LazyServicePointer<AuthorizationService> authorizationService =
       LazyServicePointer.create(
           AuthorizationService.class,
           "AuthIdentifier",
           System.getProperty("stargate.auth_id", "AuthTableBasedService"));
+  private final ServicePointer<AuthorizationProcessor> authorizationProcessor =
+      ServicePointer.create(
+          AuthorizationProcessor.class,
+          "AuthProcessorId",
+          AUTHZ_PROCESSOR_ID == null ? "NONE" : AUTHZ_PROCESSOR_ID);
```

Comment:
Why do we need this special if and return `NONE`? Is this an `OSGi` filter meaning that it won't wait for this dependent service?
I see that in the dependencies section we fiter out if null:
```
  if (AUTHZ_PROCESSOR_ID != null) {
      dependencies.add(authorizationProcessor);
    }
```
so this condition: `AUTHZ_PROCESSOR_ID == null ? "NONE" : AUTHZ_PROCESSOR_ID` shouldn't be needed.

Reply:
perhaps we could use `null` in the filter as well. It just did not feel right to me :)

Reply:
I reworked this code in more recent commits.